### PR TITLE
BigTable: Add split keys on create table - v2

### PR DESCRIPTION
**Pull request by aneepct**
Add ability to create initial splits during creation of a table. This replaces the #5370 which had rebasing issues.

### CODE REVIEW

**Comment by sduskis** [comment]
@aneepct, why did you delete all of the code in `test_utils/scripts/get_target_packages.py`?

**Comment by tseaver** [comment]
@aneepct The [failure in get_target_packages.py](https://circleci.com/gh/zakons/google-cloud-python/310?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) indicates that the `master` branch you are starting from includes stray / unrelated commits.  Please rebase your branch against the current upstream master and remove 1c72b78 from this PR.

**Comment by sduskis** [comment]
@tseaver, it looks like this PR was rebased.  Do you have time for a review?

File: bigtable/tests/system.py [hunk 1]
```diff
@@ -89,12 +89,12 @@ def setUpModule():
     if not Config.IN_EMULATOR:
         retry = RetryErrors(GrpcRendezvous,
                             error_predicate=_retry_on_unavailable)
-        instances, failed_locations = retry(Config.CLIENT.list_instances)()
+        instances_response = retry(Config.CLIENT.list_instances)()
```
Comment (tseaver) [inline bigtable/tests/system.py]: Please revert: this patch is out-of-sync with `master`.  #5420 restored returning actual `Instance` objects, rather than protos.

File: bigtable/tests/system.py [hunk 2]
```diff
@@ -119,14 +119,14 @@ def tearDown(self):
...
-
         self.assertEqual(failed_locations, [])
-        found = set([instance.name for instance in instances])
-        self.assertTrue(expected.issubset(found))
+        # We have added one new instance in `setUpModule`.
+        self.assertEqual(len(instances), len(EXISTING_INSTANCES) + 1)
+        for instance in instances:
+            instance_existence = (instance in EXISTING_INSTANCES or
+                                  instance == Config.INSTANCE)
+            self.assertTrue(instance_existence)
```
Comment (tseaver) [inline bigtable/tests/system.py]: Please revert: this patch is out-of-sync with `master`.  #5476 hardened this test against simultaneous test runs.

File: bigtable/tests/unit/test_table.py [hunk 3]
```diff
@@ -258,10 +258,11 @@ def _create_test_helper(self):
...
-        table_api = mock.create_autospec(
-            bigtable_table_admin_client.BigtableTableAdminClient)
-        instance_api = mock.create_autospec(
-            bigtable_instance_admin_client.BigtableInstanceAdminClient)
+        table_api = bigtable_table_admin_client.BigtableTableAdminClient(
+            mock.Mock())
+        instance_api = (
+            bigtable_instance_admin_client.BigtableInstanceAdminClient(
+                mock.Mock()))
         credentials = _make_credentials()
```
Comment (tseaver) [inline bigtable/tests/unit/test_table.py]: Please revert: this patch is unrelated to the rest of your change.

File: bigtable/tests/unit/test_table.py [hunk 4]
```diff
@@ -308,7 +347,7 @@ def _list_column_families_helper(self):
         from google.cloud.bigtable_admin_v2.gapic import (
             bigtable_table_admin_client)
 
-        table_api = bigtable_table_admin_client.BigtableTableAdminClient(
+        api = bigtable_table_admin_client.BigtableTableAdminClient(
```
Comment (tseaver) [inline bigtable/tests/unit/test_table.py]: Please revert: this patch is unrelated to the rest of your change.

File: bigtable/tests/unit/test_table.py [hunk 5]
```diff
@@ -324,7 +363,7 @@ def _list_column_families_helper(self):
         )
 
         # Patch the stub used by the API method.
-        client._table_admin_client = table_api
+        client._table_admin_client = api
```
Comment (tseaver) [inline bigtable/tests/unit/test_table.py]: Please revert: this patch is unrelated to the rest of your change.

File: bigtable/tests/unit/test_table.py [hunk 6]
```diff
@@ -349,8 +388,8 @@ def _read_row_helper(self, chunks, expected_result, app_profile_id=None):
             bigtable_table_admin_client)
 
         data_api = bigtable_client.BigtableClient(mock.Mock())
-        table_api = mock.create_autospec(
-            bigtable_table_admin_client.BigtableTableAdminClient)
+        table_api = bigtable_table_admin_client.BigtableTableAdminClient(
+            mock.Mock())
```
Comment (tseaver) [inline bigtable/tests/unit/test_table.py]: Please revert: this patch is unrelated to the rest of your change.

File: bigtable/tests/unit/test_table.py [hunk 7]
```diff
@@ -435,13 +474,13 @@ def test_mutate_rows(self):
         from google.cloud.bigtable_admin_v2.gapic import (
             bigtable_table_admin_client)
 
-        table_api = mock.create_autospec(
-            bigtable_table_admin_client.BigtableTableAdminClient)
+        api = bigtable_table_admin_client.BigtableTableAdminClient(
+            mock.Mock())
```
Comment (tseaver) [inline bigtable/tests/unit/test_table.py]: Please revert: this patch is unrelated to the rest of your change.

File: bigtable/tests/unit/test_table.py [hunk 8]
```diff
@@ -435,13 +474,13 @@ def test_mutate_rows(self):
...
         credentials = _make_credentials()
         client = self._make_client(project='project-id',
                                    credentials=credentials, admin=True)
         instance = client.instance(instance_id=self.INSTANCE_ID)
-        client._table_admin_client = table_api
+        client._table_admin_client = api
```
Comment (tseaver) [inline bigtable/tests/unit/test_table.py]: Please revert: this patch is unrelated to the rest of your change.

File: bigtable/tests/unit/test_table.py [hunk 9]
```diff
@@ -465,8 +504,8 @@ def test_read_rows(self):
             bigtable_table_admin_client)
 
         data_api = bigtable_client.BigtableClient(mock.Mock())
-        table_api = mock.create_autospec(
-            bigtable_table_admin_client.BigtableTableAdminClient)
+        table_api = bigtable_table_admin_client.BigtableTableAdminClient(
+            mock.Mock())
```
Comment (tseaver) [inline bigtable/tests/unit/test_table.py]: Please revert: this patch is unrelated to the rest of your change.

File: bigtable/tests/unit/test_table.py [hunk 10]
```diff
@@ -762,9 +801,9 @@ def test_callable_empty_rows(self):
         from google.cloud.bigtable_admin_v2.gapic import (
             bigtable_table_admin_client)
 
-        data_api = mock.create_autospec(bigtable_client.BigtableClient)
-        table_api = mock.create_autospec(
-            bigtable_table_admin_client.BigtableTableAdminClient)
```
Comment (tseaver) [inline bigtable/tests/unit/test_table.py]: Please revert: this patch is unrelated to the rest of your change.

File: bigtable/tests/unit/test_table.py [hunk 11]
```diff
@@ -943,8 +982,8 @@ def test_do_mutate_retryable_rows_empty_rows(self):
         from google.cloud.bigtable_admin_v2.gapic import (
             bigtable_table_admin_client)
 
-        table_api = mock.create_autospec(
-            bigtable_table_admin_client.BigtableTableAdminClient)
+        table_api = bigtable_table_admin_client.BigtableTableAdminClient(
+            mock.Mock())
```
Comment (tseaver) [inline bigtable/tests/unit/test_table.py]: Please revert: this patch is unrelated to the rest of your change.

File: bigtable/tests/system.py (hunk 1 above)
Comment (aneepct) [reply bigtable/tests/system.py]: @tseaver This is not working with GAPIC api. GAPIC api response the ListInstancesResponse instance. It gives an error ListInstancesResponse is not iterable.

**Comment by aneepct** [comment]
@tseaver I have made all the requested changes and committed them but circleci is complaining about an issue related to get_target_packages.py, which was recently changed. Can you please determine why this is creating an issue?

File: bigtable/tests/system.py (hunk 1 above)
Comment (tseaver) [reply bigtable/tests/system.py]: @aneepct You need to sync with master.  `Client.list_instances()` returns a tuple (list of `Instance`, list of failed locations).

**Comment by tseaver** [comment]
@aneepct 

> circleci is complaining about an issue related to get_target_packages.py, which was recently changed. Can you please determine why this is creating an issue?

Your branch is out of sync.  Please rebase it against the master in this repository, e.g.:

```bash
$ git remote add upstream git@github.com:GoogleCloudPlatform/google-cloud-python
$ git checkout master
$ git fetch --all
$ git merge upstream/master
$ git checkout feature/initial_split_keys_on_table
$ git rebase -i master  # remove spurious changes
$ git push -f origin feature/initial_split_keys_on_table
```

**Comment by aneepct** [comment]
@tseaver I follows your git steps it says I already rebase agains master. I also checked get_target_packages.py code with master code both are similar. Is there other files also changes with that file? That might cause an issue.

**Comment by zakons** [comment]
@sduskis Thanks for fixing the circleci build.  All the changes @tseaver requested are now committed on this branch and it should be ready for review.

**Comment by sduskis** [comment]
@tseaver, this is ready for review.

**Comment by sduskis** [comment]
@GoogleCloudPlatform/yoshi-python, this is blocked by a review.  Any chance this can be reviewed?
